components/mediation/mediators/dataservices-mediator/org.wso2.micro.integrator.mediator.dataservice/src/main/java/org/wso2/micro/integrator/mediator/dataservice/DataServiceCallMediator.java

```diff
@@ -0,0 +1,390 @@
+            if (axisService != null) {
+                if (synLog.isTraceOrDebugEnabled()) {
+                    synLog.traceOrDebug("The axisService :" + axisService.getName());
+                    synLog.traceOrDebug("The service name space URI : " + axisService.getTargetNamespace());
+                }
+                // Set the axis service into the axis2 message context
+                axis2MessageContext.setAxisService(axisService);
+                OMElement payload = addRootOperation(axis2MessageContext, messageContext);
+                if (axis2MessageContext.getEnvelope().getBody().getFirstElement() != null) {
+                    axis2MessageContext.getEnvelope().getBody().getFirstElement().detach();
+                }
+                axis2MessageContext.getEnvelope().getBody().addChild(payload);
+                dispatchToService(axis2MessageContext, messageContext);
+            } else {
+                handleException("The data service,  named '" + serviceName + "' does not exist. ", messageContext);
+            }
+
+        } catch (AxisFault axisFault) {
+            if (synLog.isTraceOrDebugEnabled())
+            {
+                synLog.traceOrDebug(axisFault.getMessage());
+            }
+            synLog.error(axisFault.getMessage());
+            handleException("AxisFault occurred.",axisFault,messageContext);
+        }
+        return true;
+    }
+    
+    private OMElement addRootOperation (org.apache.axis2.context.MessageContext axis2MessageContext,
+                                   MessageContext messageContext) {
+        Operations rootOperations = getOperations();
+        String rootOpName;
+        switch (rootOperations.getType()) {
+            case REQUEST_BOX: {
+                rootOpName = DataServiceCallMediatorConstants.REQUEST_BOX;
+                break;
+            }
+            case BATCH_REQ: {
+                Operation firstOp = (Operation) rootOperations.getOperations().get(0);
+                rootOpName = firstOp.getOperationName() + DataServiceCallMediatorConstants.BATCH_REQ_SUFFIX;
+                break;
+            }
+            case SINGLE_REQ:
+            default: {
+                Operation singleOp = (Operation) rootOperations.getOperations().get(0);
+                rootOpName = singleOp.getOperationName();
+                break;
+            }
+        }
+        QName rootOpQName = new QName(rootOpName);
+        axis2MessageContext.getAxisMessage().getAxisOperation().setName(rootOpQName);
+        OMElement payload = fac.createOMElement(rootOpName, omNamespace);
+        addOperations(rootOperations, payload, messageContext);
+        return payload;
+    }
+
+    private void dispatchToService(org.apache.axis2.context.MessageContext axis2MessageContext,
+                                   MessageContext messageContext) {
+
+        String targetTypeName = getTargetType();
+        String propertyName = getPropertyName();
```

Comment:
shall we use a different variable name here , not to hide class filed propertyName

Reply:
or we can use them directly here since class level variables .. targetType and propertyName